### ISSUE TITLE
Demo panic on Ubuntu 20.04

### ISSUE THREAD

**Issue by CallumDev** [servo/pathfinder#310]
Running the pathfinder demo in `demo/native` on latest git results in a panic with IncompatibleWinitWindow

```
RUST_BACKTRACE=full RUSTFLAGS="-C target-cpu=native" cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `/home/cmcging/src/pathfinder/target/debug/demo`
The application panicked (crashed).
Message:  called `Result::unwrap()` on an `Err` value: IncompatibleWinitWindow
Location: src/libcore/result.rs:1188
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                          (6 post panic frames hidden)                          
 6: core::result::Result<T,E>::unwrap::h9b108d71ff2f1bd6
    at /usr/src/rustc-1.41.0/src/libcore/result.rs:956
 7: demo::WindowImpl::new::h106aae37d39f3180
    at /home/cmcging/src/pathfinder/demo/native/src/main.rs:250
     248 │         window.show();
     249 │ 
     250 >         let connection = Connection::from_winit_window(&window).unwrap();
     251 │         let native_widget = connection.create_native_widget_from_winit_window(&window).unwrap();
     252 │ 
 8: demo::main::ha73be71742b38d70
    at /home/cmcging/src/pathfinder/demo/native/src/main.rs:78
      76 │     options.command_line_overrides();
      77 │ 
      78 >     let window = WindowImpl::new(&options);
      79 │     let window_size = window.size();
      80 │ 
                         (8 runtime init frames hidden)          
```

**Comment by pcwalton** [comment]
Are you running X11 or Wayland?

**Comment by CallumDev** [comment]
X11

**Comment by iamralpht** [comment]
Making this change resolves the IncompatibleWinitWindow for me, and I see the PostScript Tiger on Intel graphics with X11, but a black window on NVidia graphics with X11. The surfman threads demo behaves similarly, building with "sm-x11" gets the demo working on Intel graphics, but nets a black window with NVidia.

```
diff --git a/demo/native/Cargo.toml b/demo/native/Cargo.toml
index 39c65dc0..b3b48bfa 100644
--- a/demo/native/Cargo.toml
+++ b/demo/native/Cargo.toml
@@ -40,7 +40,7 @@ path = "../../simd"
 [dependencies.surfman]
 git = "https://github.com/pcwalton/surfman"
 rev = "616b9e020585dd650120fb1e6cceb0e2bb43199a"
-features = ["sm-winit"]
+features = ["sm-winit", "sm-x11"]
 
 [dependencies.winit]
 version = "<0.19.4" # 0.19.4 causes build errors https://github.com/rust-windowing/winit/pull/1105
```

**Comment by Darkspirit** [comment]
Yes, this can be closed as duplicate of this surfman bug:
https://github.com/servo/surfman/issues/173

**Comment by iamralpht** [comment]
@Darkspirit yes, the blank window on nvidia can (I just found that ticket!), but the pathfinder demo maybe ought to include the sm-x11 feature since X is still quite commonly used on desktop Linux systems.

**Comment by pcwalton** [comment]
This should be fixed by #323.
